File: 6-12-2024_Find H-Index.java
class Solution {
    // Function to find hIndex
    public int hIndex(int[] citations) {
         Arrays.sort(citations);
        
        int n = citations.length;
        int hIndex = 0;
        for (int i = 0; i < n; i++) {
            int papersWithAtLeastCitations = n - i;
            if (citations[i] >= papersWithAtLeastCitations) {
                hIndex = papersWithAtLeastCitations;
                break;
            }
        }
        
        return hIndex;
    }
}

File: 6-12-2024_Find H-Index.cpp
// User function Template for C++
class Solution {
  public:
    // Function to find hIndex
    int hIndex(vector<int>& citations) {
        // code here   
        int n = citations.size();
        sort(citations.begin(), citations.end());
        int ans = 0;
        for(int i=0;i<citations.size();i++) {
            if(citations[i] <= n-i) {
                ans = max(ans,citations[i]);
            }
            else {
                ans = max(ans,n-i);
            }
        }
        return ans;
    }
};